src/main/java/jarvis/storage/JsonAdaptedConsult.java

```diff
@@ -0,0 +1,72 @@
+package jarvis.storage;
+
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import jarvis.commons.exceptions.IllegalValueException;
+import jarvis.model.Consult;
+import jarvis.model.LessonAttendance;
+import jarvis.model.LessonDesc;
+import jarvis.model.LessonNotes;
+import jarvis.model.TimePeriod;
+import jarvis.model.util.SampleStudentUtil;
+
+/**
+ * Jackson-friendly version of {@link Consult}.
+ */
+public class JsonAdaptedConsult extends JsonAdaptedLesson {
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Consult's %s field is missing!";
+
+    /**
+     * Constructs a {@code JsonAdaptedConsult} with the given consult details.
+     */
+    @JsonCreator
+    public JsonAdaptedConsult(@JsonProperty("lessonDesc") String lessonDesc,
+                              @JsonProperty("timePeriod") JsonAdaptedTimePeriod timePeriod,
+                              @JsonProperty("attendance") LessonAttendance attendance,
+                              @JsonProperty("notes") LessonNotes notes,
+                              @JsonProperty("isCompleted") boolean isCompleted) {
+        super(lessonDesc, timePeriod, attendance, notes, isCompleted);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Consult toModelType() throws IllegalValueException {
+        if (this.getLessonDesc() == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
+                    LessonDesc.class.getSimpleName()));
+        }
+        if (LessonDesc.isValidLessonDesc(this.getLessonDesc())) {
+            throw new IllegalValueException(LessonDesc.MESSAGE_CONSTRAINTS);
+        }
+        final LessonDesc modelLessonDesc = new LessonDesc(this.getLessonDesc());
+
+        if (this.getTimePeriod() == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT,
+                    TimePeriod.class.getSimpleName()));
+        }
```

Comment:
Should check validity for the other fields besides `lessonDesc`. Same for other `JsonAdaptedXYZ` classes.